sky/compositor/raster_cache.cc

```diff
@@ -0,0 +1,86 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "sky/compositor/raster_cache.h"
+
+#include "sky/compositor/paint_context.h"
+#include "base/logging.h"
+#include "third_party/skia/include/core/SkImage.h"
+
+namespace sky {
+namespace compositor {
+
+static const int kRasterThreshold = 3;
+
+static bool isWorthRasterizing(SkPicture* picture) {
+  // TODO(abarth): We should find a better heuristic here that lets us avoid
+  // wasting memory on trivial layers that are easy to re-rasterize every frame.
```

Comment:
> We could time how long it takes to rasterise (vs the average time it takes to repaint things from the cache), if we have an accurate enough timer and it has a low enough overhead. That seems like the ideal algorithm. Failing that, not sure what we could do.

There's no good way to measure that because you're just inserting commands into a queue that will later be flushed.  All the time is spent in the single "flush" call.

> There's also the cost of caching things we never use again. Here we're getting around that by saving it the third time, but we should probably do an analysis of how often we use pictures, and make sure we pick a number that is a low percentile but isn't itself high.

Yep.